### DISCUSSION TITLE
Backup

### DISCUSSION THREAD

**Discussion by nep2ner** [immich-app/immich#1564] [Q&A]
I want to back up my Immich server. I know where to find the photos directory for backup as I had to specify that in the the .env file. But, how do I backup all the databases, configuration, and metadata?

**Comment by nep2ner** [comment]
Looks like I need to find the `pgdata` volume. Maybe I'm missing something obvious here.. but where is that volume located at?
https://github.com/immich-app/immich/discussions/836#discussioncomment-3918577

**Reply by jrasm91** [reply]
Just an aside, pgdata is an anonymous volume and the contents are stored (usually) in something like /var/lib/docker/..., but I usually update my compose file to use an actual folder mount similar to the upload location, ie `/home/user/docker-volumes/immich/database:/var/lib/postgresql/data`

**Reply by nep2ner** [reply]
I like that idea. Is there a way for me to change it from an anonymous volume to an actual folder mount without data loss?

**Comment by jrasm91** [comment]
You can run `docker volume inspect pgdata`

https://docs.docker.com/engine/reference/commandline/volume_inspect/

And/or `docker container inspect immich_server` to find volume information from the container.

**Comment by jpmiller25** [comment]
Docker volume backup documentation is here:
https://docs.docker.com/storage/volumes/#back-up-restore-or-migrate-data-volumes

I run a script that dumps my database you can see it here

https://github.com/immich-app/immich/discussions/1101#discussioncomment-4509303

**Reply by nep2ner** [reply]
Thanks! How do you get your `docker stop immich_postgres` command to run without `sudo` for the cron job? Or should I run the cron job as root?

**Reply by jpmiller25** [reply]
When I originally installed docker I followed the steps to add my user to the docker group by following these steps.
https://docs.docker.com/engine/install/linux-postinstall/
I'm not suggesting necessarily you should do that, check the security implications.

**Comment by MichaelBui** [comment]
IMO, you need to back up 2 things:
1. Files: <img width="583" alt="image" src="https://user-images.githubusercontent.com/1146708/216892680-cb1d7927-0887-465c-9c21-78c8a57ca752.png">
2. DB: [pg_dump](https://www.postgresql.org/docs/current/app-pgdump.html). I believe all necessary data are inside DB already: ![image](https://user-images.githubusercontent.com/1146708/216893407-33bc7997-bd14-4a0f-bd5e-0499d98887d1.png)

**Reply by bo0tzz** [reply]
@MichaelBui as an aside - it looks like you are using symlinks in your library folder? That won't work with docker - make sure the files are actually showing up where you expect them to!

**Reply by MichaelBui** [reply]
@bo0tzz yes, the symlink must be done inside the docker container first & only once (I have to login into the container using `kubectl exec` to do so) 😅
If we do symlink outside (on the host) then map the volume into the docker container, it won't work!

**Comment by nep2ner** [comment]
Do I need to backup the new tsdata that comes with v1.51.0?

**Reply by jrasm91** [reply]
No, all relevant data comes from postgres. It is more of a fancy cache server with features that make search stuff much easier to implement.

**Reply by jrasm91** [reply]
You can delete and recreate the whole container and volume at any time with just the data from postgres

**Reply by nep2ner** [reply]
Thanks!
